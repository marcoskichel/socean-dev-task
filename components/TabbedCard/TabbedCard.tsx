import { Card, CardContent, styled, Tab, Tabs } from '@mui/material';
import { ReactElement, SyntheticEvent, useState } from 'react';

interface TabData {
  label: string;
  Component: () => ReactElement;
}

interface Props {
  tabs: TabData[];
}

const StyledCard = styled(Card)(() => ({
  borderRadius: '1.5rem',
  '> div': { padding: 0 },
}));

const StyledTabs = styled(Tabs)(() => ({
  '.MuiTabs-indicator': { display: 'none' },
  '.MuiTab-root': {
    textTransform: 'none',
  },
  '.MuiTab-root[aria-selected="false"]': {
    backgroundColor: '#181818',
  },
}));

/**
 * A card with tabs
 */
const TabbedCard = (props: Props) => {
  const { tabs } = props;
  const [currentTab, setCurrentTab] = useState<number>(0);

  const handleTabChange = (event: SyntheticEvent, newValue: number) => {
    setCurrentTab(newValue);
  };

  const CurrentTabContent = tabs[currentTab]?.Component;

  return (
    <StyledCard>
      <CardContent>
        <StyledTabs
          variant="fullWidth"
          value={currentTab}
          onChange={handleTabChange}
        >
          {tabs.map((tab, index) => (
            <Tab key={index} label={tab.label} />
          ))}
        </StyledTabs>
        {CurrentTabContent && <CurrentTabContent />}
      </CardContent>
    </StyledCard>
  );
};

export default TabbedCard;
