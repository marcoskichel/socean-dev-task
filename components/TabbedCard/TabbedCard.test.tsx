import { render, screen } from '@testing-library/react';
import TabbedCard, { TabData } from './TabbedCard';

describe('TabbedCard', () => {
  const tabs: TabData[] = [
    { label: 'Tab 1', Component: () => <div>Content 1</div> },
    { label: 'Tab 2', Component: () => <div>Content 2</div> },
  ];

  it('renders', () => {
    const { asFragment } = render(<TabbedCard tabs={tabs} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the right tab', async () => {
    render(<TabbedCard tabs={tabs} selectedTab={1} />);
    const selectedTab = await screen.findByText('Content 2');
    expect(selectedTab).toBeInTheDocument();
  });
});
