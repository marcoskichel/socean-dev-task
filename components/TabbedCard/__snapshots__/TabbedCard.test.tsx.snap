// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabbedCard renders 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-cxjbia-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-lp239f-MuiCardContent-root"
    >
      <div
        class="MuiTabs-root css-1mvj89n-MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
          style="overflow: hidden; margin-bottom: 0px;"
        >
          <div
            class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth Mui-selected css-1cg5zwj-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="0"
              type="button"
            >
              Tab 1
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth css-1cg5zwj-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Tab 2
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <span
            class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
      <div>
        Content 1
      </div>
    </div>
  </div>
</DocumentFragment>
`;
